packages/wekan-oidc/oidc_server.js

```diff
@@ -1,349 +1,260 @@
     }
 }
+
 var profile = {};
 var serviceData = {};
 var userinfo = {};
 
+// Function to read the allowed emails from a local file specified in the environment variable
+var getAllowedEmailsFromFile = function() {
+    var allowedEmails = [];
+    const filePath = process.env.OAUTH2_ALLOWEDEMAILS_FILEPATH;  // Get the file path from environment variable
+
+    if (!filePath) {
+        throw new Error("OAUTH2_ALLOWEDEMAILS_FILEPATH environment variable is not set.");
+    }
+
+    try {
+        // Read the allowed emails file
+        const data = fs.readFileSync(filePath, 'utf-8');
+        allowedEmails = data.split('\n').map(email => email.trim());
+    } catch (error) {
+        console.error("Error reading allowed emails file:", error);
+    }
+    return allowedEmails;
+};
+
+// OAuth service registration
 OAuth.registerService('oidc', 2, null, function (query) {
-  var debug = process.env.DEBUG === 'true';
-
-  var token = getToken(query);
-  if (debug) console.log('XXX: register token:', token);
-
-  var accessToken = token.access_token || token.id_token;
-  var expiresAt = (+new Date) + (1000 * parseInt(token.expires_in, 10));
-
-  var claimsInAccessToken = (process.env.OAUTH2_ADFS_ENABLED === 'true'  ||
-                             process.env.OAUTH2_ADFS_ENABLED === true    ||
-                             process.env.OAUTH2_B2C_ENABLED  === 'true'  ||
-                             process.env.OAUTH2_B2C_ENABLED  === true)   || false;
-
-  if(claimsInAccessToken)
-  {
-    // hack when using custom claims in the accessToken. On premise ADFS. And Azure AD B2C.
-    userinfo = getTokenContent(accessToken);
-  }
-  else
-  {
-    // normal behaviour, getting the claims from UserInfo endpoint.
-    userinfo = getUserInfo(accessToken);
-  }
-
-  if (userinfo.ocs) userinfo = userinfo.ocs.data; // Nextcloud hack
-  if (userinfo.metadata) userinfo = userinfo.metadata // Openshift hack
-  if (debug) console.log('XXX: userinfo:', userinfo);
-
-  serviceData.id = userinfo[process.env.OAUTH2_ID_MAP]; // || userinfo["id"];
-  serviceData.username = userinfo[process.env.OAUTH2_USERNAME_MAP]; // || userinfo["uid"];
-  serviceData.fullname = userinfo[process.env.OAUTH2_FULLNAME_MAP]; // || userinfo["displayName"];
-  serviceData.accessToken = accessToken;
-  serviceData.expiresAt = expiresAt;
-
-
-  // If on Oracle OIM email is empty or null, get info from username
-  if (process.env.ORACLE_OIM_ENABLED === 'true' || process.env.ORACLE_OIM_ENABLED === true) {
-    if (userinfo[process.env.OAUTH2_EMAIL_MAP]) {
-      serviceData.email = userinfo[process.env.OAUTH2_EMAIL_MAP];
+    var debug = process.env.DEBUG === 'true';
+
+    var token = getToken(query);
+    if (debug) console.log('XXX: register token:', token);
+
+    var accessToken = token.access_token || token.id_token;
+    var expiresAt = (+new Date) + (1000 * parseInt(token.expires_in, 10));
+
+    var claimsInAccessToken = (process.env.OAUTH2_ADFS_ENABLED === 'true' ||
```

Comment:
@NadavTasher 

I don't know is it so good idea to avoid duplicate checks. Code === is used at very many places at WeKan, and duplicate check is better than explaining using == every time. Using == in general is very rare.

Reply:
@NadavTasher 

Previously at WeKan, we have used = to assing variable, and === to compare. If == is added, it gets very, very confusing, and needs a LOT of explaining.

Reply:
Although, I should change WeKan to some programming language, where is only one way to assign, and one way to compare.